internal/datastore/crdb/watch.go

```diff
@@ -105,12 +112,24 @@ func (cds *crdbDatastore) Watch(ctx context.Context, afterRevision datastore.Rev
 				return
 			}
 
-			revision, err := decimal.NewFromString(changeDetails.Updated)
+			revision, err := decimal.NewFromString(details.Updated)
 			if err != nil {
 				errs <- fmt.Errorf("malformed update timestamp: %w", err)
 				return
 			}
 
+			var caveat *core.ContextualizedCaveat
+			if details.After != nil && details.After.CaveatName != "" {
+				strct, err := structpb.NewStruct(details.After.CaveatContext)
```

Comment:
Isn't it possible to have a caveat with a name, but no context? Does NewStruct like being handed `nil`?

I think this same logic is in a couple of places

Reply:
yes, that's the case, `NewStruct` does handle well nil map. I've extended the watch test to exercise:
- caveat and nil context
- caveat and zero-value context
- caveat and context with values

I'll look into refactoring this

Reply:
added a `common.ContextualizedCaveatFrom` method